### PR TITLE
xdr: get address from signer key without panic

### PR DESCRIPTION
**Pull request by debnil**
### What

This PR adds a panic-free method to return the address from an `xdr.SignerKey` 
This PR adds a method to the `xdr.SignerKey` struct that returns the address with an error. The existing `Address` method is now a thin wrapper around this method.

### Why

In long-running, data-intensive applications, like the Hubble project, we want to avoid panics wherever possible. A panic on a single malformed input can abort the entire pipeline. A panic-free method to get the address of an `xdr.SignerKey` struct helps reduce such possible failures.

### Known limitations

N/A

### CODE REVIEW

File: xdr/signer_key.go [hunk 1]
```diff
@@ -31,10 +41,10 @@ func (skey *SignerKey) Address() string {
 		key := skey.MustPreAuthTx()
 		copy(raw, key[:])
 	default:
-		panic(fmt.Errorf("Unknown signer key type: %v", skey.Type))
+		return "", fmt.Errorf("Unknown signer key type: %v", skey.Type)
```
Comment (leighmcculloch) [inline xdr/signer_key.go]: Error strings should start with a lower case variable because they get composed together separated by colons. See #2027.
```suggestion
		return "", fmt.Errorf("unknown signer key type: %v", skey.Type)
```

File: xdr/signer_key.go [hunk 2]
```diff
@@ -10,8 +10,18 @@ import (
...
+	address, err := skey.GetAddress()
+	if err != nil {
+		panic(err)
+	}
+	return address
+}
+
+// GetAddress returns the strkey encoded form of this signer key, and an error if the
+// SignerKey is of an unknown type.
+func (skey *SignerKey) GetAddress() (string, error) {
```
Comment (leighmcculloch) [inline xdr/signer_key.go]: Can we add tests? Do we have tests for `Address()` we can duplicate for `GetAddress()` and replace an expectation for a panic with an expectation on the error?
